plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiMetadata.java

```diff
@@ -117,15 +116,25 @@ public HudiTableHandle getTableHandle(ConnectorSession session, SchemaTableName
 
     private boolean isHudiTable(ConnectorSession session, Table table)
     {
-        if (!MANAGED_TABLE.name().equals(table.getTableType()) && !EXTERNAL_TABLE.name().equals(table.getTableType())) {
-            // Views are not Hudi tables
-            return false;
+        Optional<Boolean> isHudiTable = HiveUtil.isHudiTable(table);
+        if (isHudiTable.isPresent()) {
+            if (!isHudiTable.get()) {
+                return false;
+            }
+            Location location = Location.of(table.getStorage().getLocation());
+            if (!hudiMetadataExists(this.fileSystemFactory.create(session), location)) {
+                throw new TrinoException(HUDI_BAD_DATA, "Location of table %s does not contain Hudi table metadata: %s".formatted(table.getSchemaTableName(), location));
+            }
+            return true;
         }
+
         if (table.getStorage().getOptionalLocation().isEmpty() || table.getStorage().getOptionalLocation().get().isEmpty()) {
             // No location or empty location cannot be a valid Hudi table
             return false;
         }
-        return HudiUtil.isHudiTable(fileSystemFactory.create(session), Location.of(table.getStorage().getLocation()));
+
+        Location location = Location.of(table.getStorage().getLocation());
+        return hudiMetadataExists(fileSystemFactory.create(session), location);
```

Comment:
Is it possible to refactor like below? 
```java
    private boolean isHudiTable(ConnectorSession session, Table table)
    {
        Optional<Boolean> isHudiTable = HiveUtil.isHudiTable(table);
        if (isHudiTable.isPresent() && !isHudiTable.get()) {
            return false;
        }

        if (table.getStorage().getOptionalLocation().isEmpty() || table.getStorage().getOptionalLocation().get().isEmpty()) {
            // No location or empty location cannot be a valid Hudi table
            return false;
        }

        Location location = Location.of(table.getStorage().getLocation());
        if (!hudiMetadataExists(fileSystemFactory.create(session), location)) {
            throw new TrinoException(HUDI_BAD_DATA, "Location of table %s does not contain Hudi table metadata: %s".formatted(table.getSchemaTableName(), location));
        }
        return true;
    }
```

Reply:
is the table is declared to use `HoodieParquetInputFormat`, it should be considered a hudi table, regardless whether we find `./.hoodie` inside.

Reply:
i rewrote this